app/frontend/admin/courses/StudentStanding__Root.res

```diff
@@ -0,0 +1,551 @@
+    let log = response.createUserStanding.userStanding->Belt.Option.map(userStanding => {
+      id: userStanding.id,
+      standingName: userStanding.standingName,
+      standingColor: userStanding.standingColor,
+      createdAt: userStanding.createdAt->DateFns.decodeISO,
+      creatorName: userStanding.creatorName,
+      reason: userStanding.reason,
+    })
+    switch log {
+    | Some(log) =>
+      setReason(_ => "")
+      setSelect(_ => "0")
+      setState(currentState =>
+        switch currentState {
+        | Loaded(data) =>
+          Loaded({
+            ...data,
+            userStandings: Js.Array2.concat([log], data.userStandings),
+            currentStanding: {
+              color: log.standingColor,
+              name: log.standingName,
+            },
+          })
+        | _ => currentState
+        }
+      )
+    | None => ()
+    }
+    Js.Promise.resolve()
+  })
+  ->Js.Promise2.catch(_error => {
+    setState(_ => Errored)
+    Js.Promise.resolve()
+  })
+  ->ignore
+}
+
+let updateCurrentStanding = (userStandings: userStandings, standings: standings) => {
+  switch userStandings->Js.Array2.length {
+  | 0 => {
+      let standing = Js.Array2.unsafe_get(standings, 0)
+      {
+        color: standing.color,
+        name: standing.name,
+      }
+    }
+  | _ => {
+      let standing = Js.Array2.unsafe_get(userStandings, 0)
+      {
+        color: standing.standingColor,
+        name: standing.standingName,
+      }
+    }
+  }
+}
+
+let loadStandingData = (studentId, setState) => {
+  setState(_ => Loading)
+  StudentStandingDataQuery.fetch(~notifyOnNotFound=false, {studentId: studentId})
+  ->Js.Promise2.then((response: StudentStandingDataQuery.t) => {
+    let userStandings = response.userStandings->Js.Array2.map(userStanding => {
+      id: userStanding.id,
+      standingName: userStanding.standingName,
+      standingColor: userStanding.standingColor,
+      createdAt: userStanding.createdAt->DateFns.decodeISO,
+      creatorName: userStanding.creatorName,
+      reason: userStanding.reason,
+    })
+
+    let standings = response.standings->Js.Array2.map(standing => {
+      id: standing.id,
+      name: standing.name,
+      color: standing.color,
+      description: standing.description,
+    })
+
+    setState(_ => Loaded({
+      userStandings,
+      standings,
+      currentStanding: updateCurrentStanding(userStandings, standings),
+    }))
+    Js.Promise.resolve()
+  })
+  ->Js.Promise2.catch(_error => {
+    setState(_ => Errored)
+    Js.Promise.resolve()
+  })
+  ->ignore
+}
+
+let loadPageData = (studentId, setState, setCourseId, setPageData) => {
+  SchoolAndStudentDataQuery.fetch({
+    studentId: studentId,
+  })
+  ->Js.Promise2.then((response: SchoolAndStudentDataQuery.t) => {
+    setPageData(_ => {
+      student: {
+        name: response.student.user.name,
+        email: response.student.user.email,
+      },
+      courseId: response.student.course.id,
```

Comment:
Can't you find `courseId` from the URL? Why fetch it again? 🤔

Reply:
> Can't you find courseId from the URL? 

No

Because it has only student ID and no course ID.

`http://localhost:3000/school/students/397/standing`

Reply:
Why do we need the course_id 🤔

Reply:
Is it not required for setting the course ID in the course dropdown in the nav bar? 🤔